crates/cryptography/src/witness.rs

```diff
@@ -0,0 +1,135 @@
+
+        // TODO the library should have returned the bytes, or provide a helper
+        // for conversion. For both proof and commitment
+        let commitment_x_bigint: BigUint = commitment.x.into();
+        let commitment_y_bigint: BigUint = commitment.y.into();
+
+        append_left_padded_biguint_be(&mut commitment_bytes, &commitment_x_bigint);
+        append_left_padded_biguint_be(&mut commitment_bytes, &commitment_y_bigint);
+
+        let mut proof_bytes = vec![0u8; 0];
+
+        let proof = kzg.compute_blob_proof(&input, &commitment)?;
+        let proof_x_bigint: BigUint = proof.x.into();
+        let proof_y_bigint: BigUint = proof.y.into();
+
+        append_left_padded_biguint_be(&mut proof_bytes, &proof_x_bigint);
+        append_left_padded_biguint_be(&mut proof_bytes, &proof_y_bigint);
+
+        // push data into witness
+        self.write(
+            Bytes::copy_from_slice(blob),
+            Bytes::copy_from_slice(&commitment_bytes),
+            proof_bytes.into(),
+        );
+
+        Ok(())
+    }
+
+    /// helpful function to populate struct all together
+    pub fn write(&mut self, blob: Bytes, commitment: Bytes, proof: Bytes) {
+        self.eigenda_blobs.push(blob);
+        self.commitments.push(commitment);
+        self.proofs.push(proof);
+        info!("added a blob");
+    }
+
+    /// verify all the blobs contained within the struct. It ensures the integrity of the
+    /// blob with respect the commitment. Using the batch verification method is the reduction
+    /// the size of MSM. For n blob of 4096 field element, instead of n MSM of 4096 points,
+    /// it would only require O(1) MSM of n points.
+    pub fn batch_verify(&self) -> bool {
+        info!("lib_blobs len {:?}", self.eigenda_blobs.len());
+
+        // transform to rust-kzg-bn254 inputs types
+        // TODO should make library do the parsing the return result
+        let lib_blobs: Vec<Blob> = self.eigenda_blobs.iter().map(|b| Blob::new(b)).collect();
+        let lib_commitments: Vec<G1Affine> = self
+            .commitments
+            .iter()
+            .map(|c| {
+                let x = Fq::from_be_bytes_mod_order(&c[..32]);
+                let y = Fq::from_be_bytes_mod_order(&c[32..64]);
+                G1Affine::new(x, y)
+            })
+            .collect();
+        let lib_proofs: Vec<G1Affine> = self
+            .proofs
+            .iter()
+            .map(|p| {
+                let x = Fq::from_be_bytes_mod_order(&p[..32]);
+                let y = Fq::from_be_bytes_mod_order(&p[32..64]);
+
+                G1Affine::new(x, y)
+            })
+            .collect();
+        batch::verify_blob_kzg_proof(&lib_blobs, &lib_commitments, &lib_proofs)
+            .map_err(|_| false)
+            .unwrap()
+    }
+}
+
+/// This function convert a BigUint into 32Bytes vector in big endian format
+pub fn append_left_padded_biguint_be(vec: &mut Vec<u8>, biguint: &BigUint) {
```

Comment:
hmmm doesn't feel like this function is the right abstraction. Why not just create proper serde-like functions that we can then upstream to rust-kzg-bn254?

Also this function prop shouldnt be public